component/src/lib.rs

```diff
@@ -33,81 +31,82 @@ pub trait Manageable {
 }
 
-pub struct Component<M: Manageable> {
-    shutdown_signal: oneshotSender<()>,
-    panic_signal: Receiver<IrrecoverableError>,
-    join_handle: Option<TokioJoinHandle>,
+pub struct Supervisor<M: Manageable> {
+    cancellation_signal: oneshotSender<()>,
+    irrecoverable_signal: Receiver<IrrecoverableError>,
+    join_handle: Option<JoinHandle>,
     manageable: M,
 }
 
-impl<M: Manageable + Send + Clone> Component<M> {
-    pub fn new(launcher: M) -> Self {
+impl<M: Manageable + Send + Clone> Supervisor<M> {
+    pub fn new(component: M) -> Self {
         // initialize start_fn
         // optionally, initialize the shutdown signal, the panic signal,
         // initialize join_handle to None
-        let (_, tr_panic) = channel(10);
-        let (tx_shutdown, _) = oneshotChannel();
-        Component{
-            shutdown_signal: tx_shutdown,
-            panic_signal: tr_panic,
+        let (_, tr_irrecoverable) = channel(10);
+        let (tx_cancellation, _) = oneshotChannel();
+        Supervisor {
+            cancellation_signal: tx_cancellation,
+            irrecoverable_signal: tr_irrecoverable,
             join_handle: None,
-            manageable: launcher
+            manageable: component
         }
     }
 
     // calls th launcher & stores the join handle
     async fn spawn(mut self) -> Result<(), anyhow::Error> {
-        let (tx_panic, tr_panic) = channel(10);
-        let (tx_shutdown, tr_shutdown) = oneshotChannel();
+        let (tx_irrecoverable, tr_irrecoverable) = channel(10);
+        let (tx_cancellation, tr_cancellation) = oneshotChannel();
 
         // this assumes the start_fn is populated
-        let wrapped_handle = self.manageable.start(tx_panic, tr_shutdown).await;
+        let wrapped_handle = self.manageable.start(tx_irrecoverable, tr_cancellation).await;
 
-        self.panic_signal = tr_panic;
-        self.shutdown_signal = tx_shutdown;
+        self.irrecoverable_signal = tr_irrecoverable;
+        self.cancellation_signal = tx_cancellation;
         self.join_handle = Some(wrapped_handle);
 
-        // we're ourselves a component
         self.run().await
     }
 
     // Run supervision of the child task
     async fn run(mut self) -> Result<(), anyhow::Error> {
-        // TODO : how does this conflict with join ? Do we still want join?
-        // I don't think we need to join, because we never expect these tasks to complete
-        // exception is on application shutdown, which theoretically should never happen.
-        //
         // select statement that listens for the following cases:
         //
-        // panic signal incoming => log, terminate and restart
+        // Irrecoverable signal incoming => log, terminate and restart
         // completion of the task => we're done! return
 
         if let Some(mut handle) = self.join_handle {
             loop {
                 tokio::select! {
-                    Some(message) = self.panic_signal.recv() => {
```

Comment:
^ no rules expected this token in macro call

Reply:
Yep, that's the handle polling not composing an async expression using a reference. In other words: 
`result = mut handle` should be `result = &mut handle`.